File: FileRenamer/Converters/Formats/AboutToolTipFormatConverter.cpp
#include "pch.h"
#include "AboutToolTipFormatConverter.h"
#include "AboutToolTipFormatConverter.g.cpp"

namespace winrt::FileRenamer::implementation
{
	AboutToolTipFormatConverter::AboutToolTipFormatConverter() {}

	winrt::WinrtFoundation::IInspectable AboutToolTipFormatConverter::Convert(winrt::WinrtFoundation::IInspectable const& value, winrt::WinrtInterop::TypeName const& targetType, winrt::WinrtFoundation::IInspectable const& parameter, winrt::hstring const& language)
	{
		if (value == nullptr || parameter == nullptr)
		{
			return winrt::box_value(L"");
		}

		winrt::hstring param = winrt::unbox_value<winrt::hstring>(parameter);

		if (param == L"Reference")
		{
			return winrt::box_value(winrt::unbox_value<winrt::hstring>(value) + L"\n" + AppResourceService.GetLocalized(L"About/ReferenceToolTip"));
		}
		else if (param == L"Thanks")
		{
			return winrt::box_value(winrt::unbox_value<winrt::hstring>(value) + L"\n" + AppResourceService.GetLocalized(L"About/ThanksToolTip"));
		}
		else
		{
			return value;
		}
	}
	winrt::WinrtFoundation::IInspectable AboutToolTipFormatConverter::ConvertBack(winrt::WinrtFoundation::IInspectable const& value, winrt::WinrtInterop::TypeName const& targetType, winrt::WinrtFoundation::IInspectable const& parameter, winrt::hstring const& language)
	{
		return winrt::WinrtFoundation::IInspectable();
	}
	;
}

File: FileRenamer/Converters/Formats/AboutToolTipFormatConverter.cs
using FileRenamer.Services.Root;
using System;
using Windows.UI.Xaml.Data;

namespace FileRenamer.Converters.Formats
{
    /// <summary>
    /// 关于界面项目引用和感谢介绍文字提示转换器
    /// </summary>
    public class AboutToolTipFormatConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            if (value is null || parameter is null)
            {
                return string.Empty;
            }

            string param = System.Convert.ToString(parameter);

            if (param is "Reference")
            {
                return string.Format("{0}\n{1}", value, ResourceService.GetLocalized("About/ReferenceToolTip"));
            }
            else if (param is "Thanks")
            {
                return string.Format("{0}\n{1}", value, ResourceService.GetLocalized("About/ThanksToolTip"));
            }
            else
            {
                return value;
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return default;
        }
    }
}
